test/go-chaincode/vendor-dependencies.bats

```diff
@@ -14,3 +14,124 @@ setup() {
     [ -x "${SCRIPT_DIR}/go-chaincode/vendor-dependencies.sh" ]
 }
 
+@test "vendor-dependencies.sh: fetch_dependencies should run without errors when .govendor_packages file does not exist" {
+
+    cat << EOF > sample-config.json
+{
+  "org1": {
+    "chaincode": [
+      {
+        "name": "contract1",
+        "path": "chaincode/contract1",
+        "channels": [ "channel1" ],
+        "init_args": [],
+        "instantiate": false,
+        "install": true
+      }
+    ]
+  },
+  "org2": {
+    "chaincode": [
+      {
+        "name": "contract1",
+        "path": "chaincode/contract1",
+        "channels": [ "channel1" ],
+        "init_args": [],
+        "instantiate": false,
+        "install": true
+      }
+    ]
+  }
+}
+EOF
+
+    mkdir -p "${PWD}/src/chaincode/contract1"
+    
+    stub go \
+        "get -u github.com/kardianos/govendor : true"
+
+    source "${SCRIPT_DIR}/go-chaincode/vendor-dependencies.sh"
+
+    run fetch_dependencies "sample-config.json"
+
+    # Clean up before assertions
+    rm sample-config.json
+    rm -rf "${PWD}/src/chaincode"
+    unstub go
+ 
+    # Assertions
+    [ $status -eq 0 ]  
+    [ "${lines[0]}" = "Processing org 'org1'..." ]
+    [ "${lines[1]}" = "About to fetch dependencies for 'chaincode/contract1'" ]
+    [ "${lines[3]}" = "No .govendor_packages file found; no dependencies to vendor in." ]
+
+    [ "${lines[6]}" = "Processing org 'org2'..." ]
+    [ "${lines[7]}" = "About to fetch dependencies for 'chaincode/contract1'" ]
+    [ "${lines[9]}" = "No .govendor_packages file found; no dependencies to vendor in." ]  
+}
+
+@test "vendor-dependencies.sh: fetch_dependencies should run without errors when .govendor_packages file has new lines and white spaces" {
+
+    cat << EOF > sample-config.json
+{
+  "org1": {
+    "chaincode": [
+      {
+        "name": "contract1",
+        "path": "chaincode/contract1",
+        "channels": [ "channel1" ],
+        "init_args": [],
+        "instantiate": false,
+        "install": true
+      }
+    ]
+  }
+}
+EOF
+
+    mkdir -p "${PWD}/src/chaincode/contract1"
+
+    cat << EOF > "${PWD}/src/chaincode/contract1/.govendor_packages"
+
+
+
+       github.com/hyperledger/fabric/core/chaincode/lib/cid@v1.2.1     
+
+
+
+
+  
+EOF
+    
+    stub go \
+        "get -u github.com/kardianos/govendor : true"
+
+    stub govendor \
+      "init : true"
+
+    stub govendor \
```

Comment:
Stubbing govendor twice doesn't seem right. I'm not sure how the build passed but this test is failing for me when I run it locally. I'd like to figure out what's going on before merging.

Reply:
So I think this should be something like...

```
    stub govendor \
      "init : true" \
      "fetch *github.com/hyperledger/fabric/core/chaincode/lib/cid@v1.2.1* : true" 
```

And it would be good to add a test when there's a couple of vendored packages as well

Reply:
So I think this should be something like...

```
    stub govendor \
      "init : true" \
      "fetch *github.com/hyperledger/fabric/core/chaincode/lib/cid@v1.2.1* : true" 
```

And it would be good to add a test when there's a couple of vendored packages as well.

The reason the test was failing locally for me is that I didn't have `readarray` installed. I think it would be better to use the `read` builtin instead, i.e. replace this...

```
        readarray packages < .govendor_packages
```

with this...

```
        while read package; do
            packages+=($package)
        done < .govendor_packages
```

Reply:
@jt-nti  Regrading readarray and jq, you may have missed this in the README -> https://github.com/IBM-Blockchain-Starter-Kit/build-lib/pull/88/files#diff-04c6e90faac2675aa89e2176d2eec7d8R16

There was already a dependency on jq in the codebase. readarray is available by default as long as you are using bash 4 (or higher). Less code in our codebase is less code we have to maintain, debug, update, etc. : )